### DISCUSSION TITLE
Id for HttpProxy

### DISCUSSION THREAD

**Discussion by hlange** [eclipse-edc/Connector#2104] [Q&A]
Hi,

I currently working on the "Data Plane Transfer Sync" and I was looking for an Id that the DataSource is receiving that is matching the request to the contract. 

I couldn't find any Id that could be used for this. Is the idea that this should be handled by a service outside of the EDC?

Best regards,
Henrik

**Comment by ndr-brt** [comment]
Hi, what do you mean with "the request to the contract"?
To pass parameters to the DataSource within the `data-plane-http` extension (I'm guessing you are talking about http endpoints), please take a look at the [extension documentation](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/main/extensions/data-plane/data-plane-http)

**Reply by hlange** [reply]
Hi, what I mean with "the request to the contract" is that when I request the DataSource with data-plane-http, the DataSource does not get any Information about the contract ID. This information would be helpful to monitor the processes and provide access management.

It would be helpful if the request could have the contract ID in the header when asking the DataSource.

**Reply by ndr-brt** [reply]
This is something you can actually do on the consumer side: extract the contract id from the EDR and put it into the request to the data-plane and enable query params proxy (more details in the documentation markdown I linked in my last comment)

**Reply by hlange** [reply]
Yes, but this is a bad solution:

1. I could send this information also in the body however, this means that the consumer need to add this information manually, this can be done wrong, intentionally or not
2. This information exists like you mentioned in the EDR and also in DataRequest Object when using Data Plane Transfer Client so why not forward this without the possibility of human error
3. Will be needed for the federation service Sovereign Data Exchange, so it should exist by design

**Reply by jimmarino** [reply]
Which side is the "DataSource" on, the provider, i.e., you are issuing the request directly to the provider's data plane? If so, I don't think the contract id should be passed by the consumer and trusted by the provider. An authorization header is passed and that can be used to correlate the contract id.

**Reply by hlange** [reply]
@jimmarino  I think you understand my problem. Regarding your last sentence, exist such header currently, and I didn't saw it, or should it exist and is currently not available?

**Reply by tom-rm-meyer-ISST** [reply]
I was wondering about that too. The authToken of the EDR (issued to the consumer) contains the information to validate the access via the data plane. This header is not passed to the provider's dataAsset.dataAddress.baseUrl endpoint.
Issuing the cid could allow a provider (when receiving the proxy request) to map this request to the contract and e.g. enforce policies that are not enforceable by the connector itself.

Additionally, I'm wondering if it would be beneficial to be able to route through DAPS information. E.g. in Catena-X I would be able to receive verified partner identification data I would otherwise need to pass through a header / body as a consumer, which is somewhat contradictionary to the trust provided by the connector.

**Reply by jimmarino** [reply]
The EDR token contains the CID as a claim, see `ConsumerPullTransferTokenDecorator`.
